src/compiler/transform/compiled/create-appendix.ts

```diff
@@ -0,0 +1,115 @@
+// TODO: Refactor, use Rollup parser (ESTree AST) to transform the output.
+
+import url from 'node:url';
+import MagicString from 'magic-string';
+
+import { getNameFromStoryAttribute } from '../../../parser/analyse/Story-attributes.js';
+import type { CompiledASTNodes } from '../../../parser/extract/compiled/nodes.js';
+import type { SvelteASTNodes } from '../../../parser/extract/svelte/nodes.js';
+import { extractStoryAttributesNodes } from '../../../parser/extract/svelte/Story-attributes.js';
+
+const STORY_FNS_VARIABLE = '__storyFns';
+const EXPORT_ORDER_VARIABLE = '__namedExportsOrder';
+
+interface Params {
+  componentName: string;
+  code: MagicString;
+  nodes: {
+    compiled: CompiledASTNodes;
+    svelte: SvelteASTNodes;
+  };
+  filename: string;
+}
+
+export async function createAppendix(params: Params) {
+  const { componentName, code, nodes, filename } = params;
+  const { compiled, svelte } = nodes;
+  const { defineMetaVariableDeclaration } = compiled;
+
+  // NOTE:
+  // We need to remove the default export from the code,
+  // because Storybook internally expects export default `meta`
+  code.replace(/export default /, '');
+
+
+  const exportsOrder = await getStoriesNames({ nodes: svelte, filename });
+
+  const storiesExports = exportsOrder
+    .map((name) => {
+      // TODO: There's probably some internal function in the Storybook to handle this?
```

Comment:
there is not, only the other way around (from export name to story name or story id).

But I think we need a consistent way to go from name to id and export name:

Eg.

- Name: "My cool! story name"
- Id: "my-cool-story-name"
- Export: "MyCoolStoryName"

This can cause collisions that we need to handle gracefully. The current version of this addon allows passing a custom id, but that is inconsistent with regular CSF which doesn't have this, but instead supports setting a custom export name. I'm leaning towards we want to support that, as it also makes it easier to import stories in MDX files.

I've taken an initial stab at this in `utils/identifiers.ts`, but haven't actually used them anywhere yet.